hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBasicRetryCondition.java

```diff
@@ -7,7 +7,7 @@
 public class HermesClientBasicRetryCondition implements Predicate<HermesResponse> {
     @Override
     public boolean test(HermesResponse response) {
-        return isClientTimeoutOrServerError(response) || isFailedExceptionally(response);
+        return (response != null) && (isClientTimeoutOrServerError(response) || isFailedExceptionally(response));
```

Comment:
Issue was caused by an exception from discovery service (no consul service on host).

There is a separate mechanism that repeats requests in case of exceptions, but anyway I've changed the predicate according to your suggestion.